core/templates/pages/exploration-editor-page/translation-tab/services/translation-language.service.ts

```diff
@@ -17,38 +17,52 @@
  * in the translation tab is currently active.
  */
 
-import { EventEmitter } from '@angular/core';
-
-require('domain/utilities/language-util.service.ts');
-
-angular.module('oppia').factory('TranslationLanguageService', [
-  '$log', 'LanguageUtilService',
-  function($log, LanguageUtilService) {
-    var activeLanguageCode = null;
-    var allAudioLanguageCodes = (
-      LanguageUtilService.getAllVoiceoverLanguageCodes());
-    var _activeLanguageChangedEventEmitter = new EventEmitter();
-    return {
-      getActiveLanguageCode: function() {
-        return activeLanguageCode;
-      },
-      setActiveLanguageCode: function(newActiveLanguageCode) {
-        if (allAudioLanguageCodes.indexOf(newActiveLanguageCode) < 0) {
-          $log.error('Invalid active language code: ' + newActiveLanguageCode);
-          return;
-        }
-        activeLanguageCode = newActiveLanguageCode;
-        _activeLanguageChangedEventEmitter.emit();
-      },
-      getActiveLanguageDescription: function() {
-        if (!activeLanguageCode) {
-          return null;
-        }
-        return LanguageUtilService.getAudioLanguageDescription(
-          activeLanguageCode);
-      },
-      get onActiveLanguageChanged() {
-        return _activeLanguageChangedEventEmitter;
-      }
-    };
-  }]);
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { EventEmitter, Injectable } from '@angular/core';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+import { LoggerService } from 'services/contextual/logger.service';
+import { List } from 'lodash';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TranslationLanguageService {
+  constructor(
+    private languageUtilService: LanguageUtilService,
+    private loggerService: LoggerService) {}
+
+  private activeLanguageCode: string = null;
+  private allAudioLanguageCodes: List<string> = (
+    this.languageUtilService.getAllVoiceoverLanguageCodes());
+  private _activeLanguageChangedEventEmitter = new EventEmitter<void>();
+
+  getActiveLanguageCode = function(): string {
```

Comment:
Here and everywhere else in this file.
```suggestion
  getActiveLanguageCode(): string {
```